src/main/kubeconfig-manager.ts

```diff
@@ -24,14 +24,24 @@ export class KubeconfigManager {
   protected async init() {
     try {
       await this.contextHandler.ensurePort();
-      await this.createProxyKubeconfig();
+      this.tempFile = this.createProxyKubeconfig();
     } catch (err) {
       logger.error(`Failed to created temp config for auth-proxy`, { err });
     }
   }
 
   getPath() {
+    // create proxy kubeconfig if it is removed
+    if (this.tempFile && !fs.pathExistsSync(this.tempFile)) {
+      try {
+        this.tempFile = this.createProxyKubeconfig();
+      } catch (err) {
+        logger.error(`Failed to created temp config for auth-proxy`, { err });
+      }
```

Comment:
If there is a problem with the temp files being removed, shouldn't it be better if we store the files under `userData` instead, clearing them out on every "remove cluster" and "start Lens" event?

Reply:
probably both fixes make sense.